File: ws/first_ws/src/topic_pkg/scripts/custom_msg_obj_publisher.py
#! /usr/bin/env python
# coding:utf8

import rospy
from demo_msgs.msg import Student, Team

if __name__ == '__main__':
    nodeName = "publish_custom_msg_obj_py_node"
    rospy.init_node(nodeName)

    topicName = "/custom/msg/obj/topic"
    pub = rospy.Publisher(topicName, Team, queue_size=1000)

    rate = rospy.Rate(1)
    team = Team()
    stu = Student()

    idx = 0
    while not rospy.is_shutdown():
        team.name = "NCHU %d" % idx

        stu.name = "tom %d" % idx
        stu.age = idx
        team.leader = stu

        pub.publish(team)

        idx += 1
        rate.sleep()

    rospy.spin()

File: 06ros/ws01/first_ws/src/topic_pkg/src/custom_msg_obj_publisher.cpp
//
// Created by zcb on 2021/4/2.
//
#include <iostream>
#include <ros/ros.h>
#include <demo_msgs/Team.h>

using namespace std;

int main(int argc, char **argv) {
    string nodeName = "publish_custom_msg_obj_cpp_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    string topicName = "/custom/msg/obj/topic";
    ros::Publisher pub = node.advertise<demo_msgs::Team>(topicName, 1000);

    demo_msgs::Team team;
    demo_msgs::Student stu;
    ros::Rate rate(1);
    int idx = 0;
    while (ros::ok()) {
        team.name = "NCHU" + to_string(idx);
        stu.name = "tom " + to_string(idx);
        stu.age = idx;
        team.leader = stu;
        pub.publish(team);

        idx++;
        rate.sleep();
    }
    ros::spin();

    return 0;
}







